px4_ros2_cpp/test/unit/shared_subscription.cpp

```diff
@@ -126,34 +131,40 @@ TEST_F(VehicleStatusTest, State) {
   EXPECT_EQ(message_counters[2], 10U);
 }
 
-TEST_F(VehicleStatusTest, MultiInstance) {
-  SharedVehicleStatus & vehicle_status1 = SharedVehicleStatus::instance(*_node);
+TEST_F(SharedSubscriptionTest, MultiInstance) {
+  const std::string topic_name = "/unit_test/fmu/out/vehicle_status";
+  auto & vehicle_status1 = SharedSubscription<px4_msgs::msg::VehicleStatus>::instance(
+    *_node,
+    topic_name);
   auto vehicle_status_pub1 = _node->create_publisher<px4_msgs::msg::VehicleStatus>(
-    "/fmu/out/vehicle_status" + px4_ros2::getMessageNameVersion<px4_msgs::msg::VehicleStatus>(), rclcpp::QoS(
+    topic_name, rclcpp::QoS(
       1).best_effort());
 
-  const std::string topic_prefix = "test/";
-  SharedVehicleStatus & vehicle_status2 = SharedVehicleStatus::instance(*_node, topic_prefix);
+  const std::string topic_prefix = "test";
+  auto & vehicle_status2 = SharedSubscription<px4_msgs::msg::VehicleStatus>::instance(
+    *_node,
+    topic_prefix +
+    topic_name);
   auto vehicle_status_pub2 = _node->create_publisher<px4_msgs::msg::VehicleStatus>(
-    topic_prefix + "fmu/out/vehicle_status" + px4_ros2::getMessageNameVersion<px4_msgs::msg::VehicleStatus>(), rclcpp::QoS(
+    topic_prefix + topic_name, rclcpp::QoS(
       1).best_effort());
 
   ASSERT_NE(&vehicle_status1, &vehicle_status2);
 
   px4_msgs::msg::VehicleStatus pub_msg;
-  std::array<std::optional<SharedVehicleStatusToken>, 2> tokens;
+  std::array<std::optional<SharedSubscriptionCallbackInstance>, 2> instances;
```

Comment:
```suggestion
  std::array<std::optional<SharedSubscriptionCallbackInstance>, 2> callbacks;
```
very minor nitpick, but i find the term `instance` a bit confusing in these tests because they can both refer to instances of subscriptions or callbacks. And the test is called `MultiInstance` presumably refering to subscriptions in that case